### ISSUE TITLE
i18n: Language selector

### ISSUE THREAD

**Issue by olizilla** [ipfs-shipyard/ipfs-webui#795]
The new webui is using the value returned from `navigator.language` to pick the locale to load. There is no way to override that from the UI. It'd be great to have a language selector that lets you change the current locale manually.

This would expose the number of languages that the UI is available in, which is a nice side effect, and would give us a place to add a "contribute a translation" call to action to get more eyes on https://www.transifex.com/ipfs/

This would also help with developing and testing the UI in different languages. Setting up multiple profiles for different languages is a faff.

**Comment by olizilla** [comment]
@fsdiogo we need to sketch out what this should look like, if you have any thoughts, or can grab some good examples from elsewhere that'd be rad. Maybe, just maybe, there is already a good react component for this floating around out there in npmspace somewhere.

**Comment by olizilla** [comment]
yep, I think ideally we'd get the language name in the local language instead of the shortcode. The shortcode is always in ascii, so isn't super friendly to non-latin-alphabet users. Would be good to get some screenshots of some existing language selectors that do this well. Should we include flags or are they divisive in some locales?

Can we just expose or query the list of locales we set up here? https://github.com/ipfs-shipyard/ipfs-webui/blob/1774658fb277ea6b6d32238b43d0a1f00f3919bb/src/i18n.js#L26

**Comment by fsdiogo** [comment]
Yeah, we should have the language name instead of the shortcode for sure!

I've been poking around a bit and to my surprise the dropdown isn't the tool of choice, but rather the modal:

|Google|Dropbox|Soundcloud|
|------|-------|----------|
|![google](https://user-images.githubusercontent.com/33324750/47080847-fceca400-d200-11e8-8d4e-e55664e166b9.gif)|![dropbox](https://user-images.githubusercontent.com/33324750/47080848-fceca400-d200-11e8-8e2e-bd948748c874.gif)|![soundcloud](https://user-images.githubusercontent.com/33324750/47080849-fd853a80-d200-11e8-8702-83abe5f7d33f.gif)|

An explanation to this can be the vast number of languages supported, in a dropdown it wouldn't be that much user friendly.

I didn't think the flags were divisive and I thought we should exclude them, but after scanning through a few articles I'm not sure anymore:
- https://usersnap.com/blog/design-language-switch/
- http://www.flagsarenotlanguages.com/blog/best-practice-for-presenting-languages/
- http://daily.unitedlanguagegroup.com/stories/editorials/inside-design-language-selector-no-flags


> Can we just expose or query the list of locales we set up here?

Yeah, that should do it!
